docs/mdx-components.tsx

```diff
@@ -52,7 +52,7 @@ function Link({
 }
 
 interface MDXComponentProps {
-  [key: string]: (props: unknown) => JSX.Element;
+  [key: string]: (props: unknown) => React.JSX.Element;
```

Comment:
Ok wow, didn't realize the global namespace was deprecated and I thought it was part of the global typescript interfaces (discovered the deprecation here):  https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f5b0d6b088a3b1ff5b74dc33ef044c81c478aeae/types/react/index.d.ts#L4302-L4305 . We should ensure this is backwards compatible with 16/18.

Reply:
Looks like they added support for react 16 and 17.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v16/index.d.ts#L3136-L3139
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/index.d.ts#L3136-L3139